File: public/examples/integrations/toolkits/salesforce/get_account_data_by_keywords_example_llm_oai.py
import os
from openai import OpenAI

PROMPT = "Get the data for the account with the name 'Acme Inc'"
TOOL_NAME = "Salesforce.GetAccountDataByKeywords"

client = OpenAI(
    base_url="http://localhost:9099/v1", api_key=os.environ.get("ARCADE_API_KEY")
)

response = client.chat.completions.create(
    messages=[
        {"role": "user", "content": PROMPT},
    ],
    model="gpt-4o-mini",
    tools=[TOOL_NAME],
    tool_choice="generate",
)
print(response.choices[0].message.content)

File: public/examples/integrations/toolkits/salesforce/get_account_data_by_keywords_example_llm_oai.js
import OpenAI from 'openai';

const PROMPT = "Get the data for the account with the name 'Acme Inc'";
const TOOL_NAME = "Salesforce.GetAccountDataByKeywords";

const client = new OpenAI({
  baseURL: 'http://localhost:9099/v1',
  apiKey: process.env.ARCADE_API_KEY
});

const response = await client.chat.completions.create({
  messages: [
    { role: 'user', content: PROMPT }
  ],
  model: 'gpt-4o-mini',
  tools: [TOOL_NAME],
  tool_choice: 'generate'
});

console.log(response.choices[0].message.content);
